### ISSUE TITLE
Add Persistence

### ISSUE THREAD

**Issue by jsmnbom** [python-telegram-bot/python-telegram-bot#638]
It would be lovely to have some sort of persistence in the library. 
We've talked in the dev group and we'd like it to look something like this:
``` python
updater = Updater('Token')  # Fires UserWarning('No persistence bla bla, use persistence=None to suppress')
updater = Updater('Token', persistence=None)
updater = Updater('Token', persistence=SQLitePersistence('data.sqlite', autocommit=False))
updater = Updater('Token', persistence=SQLitePersistence('data.sqlite', autocommit=True))
```
We should probably start with a FilePersistence and then if needed expand into SQLite or whatever.

If you have any questions and/or question about how you'd like us to implement this, please comment below :)

**Comment by patrnk** [comment]
Every bot I've made was running in several processes. I've heard [SQLite is bad at concurrent access](https://stackoverflow.com/questions/1063438/sqlite3-and-multiple-processes). So wouldn't it be a problem?

**Comment by bomjacob** [comment]
If you don't mind me asking... how are you running your bot in several processes? How are you sharing resources between them?
I'm thinking that we start by focusing on getting the feature implemented with a disk cache (using pickle or something). In that first stage, it's not gonna be multiprocess compatible for sure, but as we get other persistence methods (SQLite or other database) we'll look into it.

**Comment by patrnk** [comment]
I've always thought of uwsgi workers as separate processes, and `'data.sqlite'` file would be the shared resource. Isn't it?

You've answered my question, thank you.

**Comment by jsmnbom** [comment]
Closed by #1017
